src/pages/policies/pagination.tsx

```diff
@@ -73,60 +71,56 @@ export default function Pagination(props: Props) {
 }
 
 const MIN_PAGE_COUNT = 9;
-
-export function usePagination({
-  totalResults,
-  currentPage,
-  pageSize = 10,
-}: {
-  totalResults: number;
-  currentPage: number;
-  pageSize?: number;
-}): Array<number | '...'> {
-  return useMemo(() => {
-    const totalPageCount = Math.ceil(totalResults / pageSize);
-
-    if (totalPageCount < MIN_PAGE_COUNT) {
-      return getNumberRange(1, totalPageCount);
-    }
-
-    const lastPage = totalPageCount;
-
-    const previousPage = Math.max(currentPage - 1, 1);
-    const showDotsOnLeft = previousPage >= 4;
-
-    const nextPage = Math.min(currentPage + 1, lastPage);
-    const showDotsOnRight = nextPage <= lastPage - 3;
-
-    // 1 2 3 4 5 ... 9 10
-    if (!showDotsOnLeft && showDotsOnRight) {
-      return [...getNumberRange(1, 5), '...', lastPage - 1, lastPage];
-    }
-
-    // 1 2 ... 6 7 8 9 10
-    if (showDotsOnLeft && !showDotsOnRight) {
-      return [1, 2, '...', ...getNumberRange(lastPage - 4, lastPage)];
-    }
-
-    // 1 ... 5 6 7 ... 10
-    return [1, '...', ...getNumberRange(previousPage, nextPage), '...', lastPage];
-  }, [totalResults, pageSize, currentPage]);
+const DEFAULT_PAGE_SIZE = 10;
+
+export function paginate(
```

Comment:
I'd like to use pagination in the governance section as well to make proposals loading more performant. For that we would need probably a bit more generic version, but I don't mind leaving that for a bit later.

I think for now most people use https://enormous.cloud/dao/api3/tracker/votings to browse past proposals, because on DAO dashboard it loads really slow.

Reply:
Re: your comment about the proposals, I would think the appropriate mechanism to drive paging would also be a query param, so this component would probably be generic enough already as it doesn't hardcode the path:
```tsx
return location.pathname + '?' + newParams.toString();
```